scripts/Paddle.gd

```diff
@@ -24,3 +28,6 @@ func _physics_process(delta):
 			direction += Vector2(1, 0)
 	
 		position += direction.normalized() * speed * delta
+
+func on_collide_with_ball(ball, collision):
+	ball.tweak_direction((collision.get_normal() * 2) + direction)
```

Comment:
favour up/down twice as much as the direction of the paddle with `collision.get_normal() * 2`

also we need get_normal instead of just Vector2.UP because if the ball hits the side of the paddle it gets stuck

Reply:
Nice! The paddle should perhaps try and use this logic to line up shots